dlt/destinations/impl/clickhouse/sql_client.py

```diff
@@ -0,0 +1,209 @@
+            return None if curr.description is None else curr.fetchall()
+
+    def create_dataset(self) -> None:
+        # Clickhouse doesn't have schemas.
+        pass
+
+    def drop_dataset(self) -> None:
+        # Since Clickhouse doesn't have schemas, we need to drop all tables in our virtual schema,
+        # or collection of tables, that has the `dataset_name` as a prefix.
+        to_drop_results = self.execute_sql(
+            """
+            SELECT name
+            FROM system.tables
+            WHERE database = %s
+            AND name LIKE %s
+            """,
+            (
+                self.database_name,
+                f"{self.dataset_name}%",
+            ),
+        )
+        for to_drop_result in to_drop_results:
+            table = to_drop_result[0]
+            # The "DROP TABLE" clause is discarded if we allow clickhouse_driver to handle parameter substitution.
+            # This is because the driver incorrectly substitutes the entire query string, causing the "DROP TABLE" keyword to be omitted.
+            # To resolve this, we are forced to provide the full query string here.
+            self.execute_sql(
+                f"""DROP TABLE {self.capabilities.escape_identifier(self.database_name)}.{self.capabilities.escape_identifier(table)} SYNC"""
+            )
+
+    @contextmanager
+    @raise_database_error
+    def execute_query(
+        self, query: AnyStr, *args: Any, **kwargs: Any
+    ) -> Iterator[ClickhouseDBApiCursorImpl]:
+        assert isinstance(query, str), "Query must be a string"
+
+        db_args = kwargs.copy()
+
+        if args:
+            query, db_args = _convert_to_old_pyformat(query, args, OperationalError)
+            db_args.update(kwargs)
+
+        with self._conn.cursor() as cursor:
+            for query_line in query.split(";"):
+                if query_line := query_line.strip():
+                    try:
+                        cursor.execute(query_line, db_args)
+                    except KeyError as e:
+                        raise DatabaseTransientException(OperationalError()) from e
+
+            yield ClickhouseDBApiCursorImpl(cursor)  # type: ignore[abstract]
+
+    def fully_qualified_dataset_name(self, escape: bool = True) -> str:
+        database_name = (
+            self.capabilities.escape_identifier(self.database_name)
+            if escape
+            else self.database_name
+        )
+        dataset_name = (
+            self.capabilities.escape_identifier(self.dataset_name) if escape else self.dataset_name
+        )
+        return f"{database_name}.{dataset_name}"
+
+    def make_qualified_table_name(self, table_name: str, escape: bool = True) -> str:
+        database_name = (
+            self.capabilities.escape_identifier(self.database_name)
+            if escape
+            else self.database_name
+        )
+        dataset_table_name = (
+            self.capabilities.escape_identifier(f"{self.dataset_name}_{table_name}")
```

Comment:
we should make the separator between dataset name and table name configurable and set it to something special, so we don't end up dropping tables by mistake because you are deleting a dataset that is also contained in the name of another dataset that exists, do you know what I mean? maybe the default dataset default separator should be three underscores or something like this.

Reply:
Added configuration option `[destination.clickhouse.credentials.dataset_table_separator]`